cmd/influxd/run/config.go

```diff
@@ -175,6 +175,9 @@ func (c *Config) applyEnvOverrides(prefix string, spec reflect.Value) error {
 			// e.g. GRAPHITE_0
 			if f.Kind() == reflect.Slice || f.Kind() == reflect.Array {
 				for i := 0; i < f.Len(); i++ {
+					if err := c.applyEnvOverrides(key, f.Index(i)); err != nil {
```

Comment:
This code specifically seems to be wrong. This code is confusing so I'm not certain, but I think this would allow me to say `INFLUXDB_0_BIND_ADDRESS` to set all of the bind addresses in every section where that applies. If I'm reading this right, the unit test you have should work with the original code.